### ISSUE TITLE
project run with jacoco and quasar, quasar does not work

### ISSUE THREAD

**Issue by huxinxinxinxin** [jacoco/jacoco#624]
spring-boot 1.5.8.RELEASE
quasar 0.7.6
jacoco 0.7.10
jdk 1.8

Hi,
project run with quasar alone, work well
but run with jacoco and quasar, quasar does not work. I get a error:
```
WARNING: Uninstrumented methods (marked '**') or call-sites (marked '!!') detected on the call stack:
at co.paralleluniverse.common.util.ExtendedStackTrace.here (ExtendedStackTrace.java:44 bci: 12)
at co.paralleluniverse.fibers.Fiber.checkInstrumentation (Fiber.java:1668 bci: 4)
at co.paralleluniverse.fibers.Fiber.verifySuspend (Fiber.java:1641 bci: 25)
at co.paralleluniverse.fibers.Fiber.verifySuspend (Fiber.java:1636 bci: 7)
at co.paralleluniverse.fibers.Fiber.park (Fiber.java:620 bci: 5)
at co.paralleluniverse.fibers.Fiber.park (Fiber.java:624 bci: 8)
at co.paralleluniverse.fibers.FiberAsync.run) (FiberAsync.java:121 bci: 244)
at co.paralleluniverse.fibers.httpclient.AsyncHttpReq.run) (AsyncHttpReq.java:26 bci: 94)
at co.paralleluniverse.fibers.httpclient.FiberHttpClient.doExecute(org.apache.http.HttpHost,org.apache.http.HttpRequest,org.apache.http.protocol.HttpContext) (FiberHttpClient.java:96 bci: 201)
at co.paralleluniverse.fibers.httpclient.FiberHttpClient.execute(org.apache.http.client.methods.HttpUriRequest,org.apache.http.protocol.HttpContext) (FiberHttpClient.java:161 bci: 194)
at co.paralleluniverse.fibers.httpclient.FiberHttpClient.execute(org.apache.http.client.methods.HttpUriRequest,org.apache.http.protocol.HttpContext) (FiberHttpClient.java:47 bci: 173)
at org.springframework.http.client.HttpComponentsClientHttpRequest.executeInternal(org.springframework.http.HttpHeaders,byte[]) (HttpComponentsClientHttpRequest.java:89 bci: 83)
** at org.springframework.http.client.AbstractBufferingClientHttpRequest.executeInternal (AbstractBufferingClientHttpRequest.java:48 bci: 51)
** at org.springframework.http.client.AbstractClientHttpRequest.execute (AbstractClientHttpRequest.java:53 bci: 18) 
** at org.springframework.http.client.BufferingClientHttpRequestWrapper.executeInternal (BufferingClientHttpRequestWrapper.java:56 bci: 42) 
** at org.springframework.http.client.AbstractBufferingClientHttpRequest.executeInternal (AbstractBufferingClientHttpRequest.java:48 bci: 51) 
** at org.springframework.http.client.AbstractClientHttpRequest.execute (AbstractClientHttpRequest.java:53 bci: 18) 
** at org.springframework.http.client.InterceptingClientHttpRequest$InterceptingRequestExecution.execute (InterceptingClientHttpRequest.java:99 bci: 1333)
at com.sc.integration.config.ScClientHttpRequestInterceptor.intercept (ScClientHttpRequestInterceptor.java:46 bci: 239) 
** at org.springframework.http.client.InterceptingClientHttpRequest$InterceptingRequestExecution.execute (InterceptingClientHttpRequest.java:86 bci: 275) 
at org.springframework.http.client.InterceptingClientHttpRequest.executeInternal (InterceptingClientHttpRequest.java:70 bci: 22) 
** at org.springframework.http.client.AbstractBufferingClientHttpRequest.executeInternal (AbstractBufferingClientHttpRequest.java:48 bci: 51) 
** at org.springframework.http.client.AbstractClientHttpRequest.execute (AbstractClientHttpRequest.java:53 bci: 18) 
** at org.springframework.web.client.RestTemplate.doExecute (RestTemplate.java:652 bci: 80) 
** at org.springframework.web.client.RestTemplate.execute (RestTemplate.java:613 bci: 33) 
** at org.springframework.web.client.RestTemplate.exchange (RestTemplate.java:531 bci: 43) 
** at com.social.credits.integration.crawler.NewsCommonResourceImpl.getNewsDetailData (NewsCommonResourceImpl.java:42 bci: 70) 
** at com.social.credits.common.util.ScChannelUtils.lambda$goParamFunc1ChannelOptional$c11c868e$1 (ScChannelUtils.java:147 bci: 6) 
** at co.paralleluniverse.strands.SuspendableUtils$VoidSuspendableCallable.run (SuspendableUtils.java:44 bci: 8) 
at co.paralleluniverse.strands.SuspendableUtils$VoidSuspendableCallable.run (SuspendableUtils.java:32 bci: 5) 
at co.paralleluniverse.fibers.Fiber.run (Fiber.java:1072 bci: 21) 
at co.paralleluniverse.fibers.Fiber.run1 (Fiber.java:1067 bci: 5)
```

**Comment by marchof** [comment]
@huxinxinxinxin Hard to tell what the problem here is. Looks like Quasar also performs some kind of instrumentation, as this is a Quasar message:

    WARNING: Uninstrumented methods (marked '**') or call-sites (marked '!!') detected on the call stack: 

We have no experience with quasar on our side. If you want us to look at it, please provide a minimal reproducer (e.g. GitHub repo) that we can execute and analyze.

**Comment by huxinxinxinxin** [comment]
Thanks, I will provide a demo later,
ps:  idea coverage plugin, Emma is work well with quasar

**Comment by huxinxinxinxin** [comment]
@marchof 
demo : https://github.com/huxinxinxinxin/jacocodemo

**Comment by Godin** [comment]
@huxinxinxinxin could you please explain what to do with this project to reproduce the problem? `mvn clean package` doesn't show any errors/warnings.

**Comment by huxinxinxinxin** [comment]
This error occurs at run time

**Comment by huxinxinxinxin** [comment]
You can run the idea run application with coverage, select jacoco, then visit localhost:8080/api/demo, you can see this error

**Comment by huxinxinxinxin** [comment]
if run jar, you need jacocoagent.jar Configure to mvn, and modify Java-Agents.
after 'mvn clean install', 
then run demo-cp.jar.'java -jar demo-cp.jar' ,
visit localhost:8080/api/demo, you can see this error

**Comment by Godin** [comment]
@huxinxinxinxin 

> You can run the idea run application

If this means execution from IntelliJ IDEA, then it fails even without coverage:

```
QUASAR WARNING: Quasar Java Agent isn't running. If you're using another instrumentation method you can ignore this message; otherwise, please refer to the Getting Started section in the Quasar documentation.
Exception in Fiber "fiber-10000001" java.lang.AssertionError
```

> if run jar, you need jacocoagent.jar Configure to mvn, and modify Java-Agents.

please provide **complete** example that doesn't require specific IDE or specific IDE configuration and that doesn't require additional modifications and **easily executable** from command line.

**Comment by huxinxinxinxin** [comment]
sorry. i forget this
idea need add jvm :-javaagent:/home/hx-pc/.m2/repository/co/paralleluniverse/quasar-core/0.7.6/quasar-core-0.7.6-jdk8.jar=m -Dco.paralleluniverse.fibers.verifyInstrumentation=true 
sorry again.
mvnrepository  can only find org.jacoco.agent and org.jacoco.core, don`t have jacocoagent.jar.
so  jacocoagent.jar own upload to the local mvnrepository configuration

**Comment by Godin** [comment]
> mvnrepository can only find org.jacoco.agent and org.jacoco.core, don`t have jacocoagent.jar

As explained in documentation on page http://www.jacoco.org/jacoco/trunk/doc/repo.html agent is available in Maven Central repository:

> Group ID | Artifact ID | Classifier | Description
> -- | -- | -- | --
> org.jacoco | org.jacoco.agent | runtime | Agent

Using following `patch.diff` file

```diff
--- pom.xml
+++ pom.xml
@@ -39,7 +39,7 @@
         <dependency>
             <groupId>co.paralleluniverse</groupId>
             <artifactId>quasar-core</artifactId>
-            <version>0.7.6</version>
+            <version>0.7.9</version>
             <classifier>jdk8</classifier>
         </dependency>
         <dependency>
@@ -53,9 +53,10 @@
             <version>0.7.0</version>
         </dependency>
         <dependency>
-            <groupId>org.jacocoagent</groupId>
-            <artifactId>jacocoagent</artifactId>
-            <version>sc-version</version>
+            <groupId>org.jacoco</groupId>
+            <artifactId>org.jacoco.agent</artifactId>
+            <classifier>runtime</classifier>
+            <version>0.7.9</version>
         </dependency>
     </dependencies>
     <build>
@@ -66,7 +67,7 @@
                 <artifactId>capsule-maven-plugin</artifactId>
                 <version>1.3.0</version>
                 <configuration>
-                    <appClass>com.social.credits.Application</appClass>
+                    <appClass>com.demo.Application</appClass>
                     <type>fat</type>

                     <execPluginConfig>root</execPluginConfig>
@@ -74,15 +75,15 @@

                     <properties>
                         <property>
-                            <key>propertyName1</key>
-                            <value>propertyValue1</value>
+                            <key>co.paralleluniverse.fibers.verifyInstrumentation</key>
+                            <value>true</value>
                         </property>
                     </properties>

                     <manifest>
                         <entry>
                             <key>Java-Agents</key>
-                            <value>quasar-core-0.7.6-jdk8.jar=m jacocoagent-sc-version.jar=output=tcpserver,port=18080,address=127.0.0.1,append=false</value>
+                            <value>org.jacoco.agent-0.7.9-runtime.jar quasar-core-0.7.9-jdk8.jar=m</value>
                         </entry>
                     </manifest>
```

and following commands

```
$ git clone git@github.com:huxinxinxinxin/jacocodemo.git
$ cd jacocodemo
$ git rev-parse HEAD
f39bd6e08ad1dd76f4c1a6e6e1a84796be4c3fd2
$ patch -p0 < ../patch.diff
$ mvn clean package
$ java -jar target/demo-cp.jar
```

application starts without any exceptions and responds like with and without JaCoCo:

```
$ curl http://localhost:8080/api/demo
{"from":"zh","to":"en","trans_result":[{"src":"中国","dst":"China"}]}
```

At a termination of application JaCoCo exec file is populated as expected:

```
$ du -h jacoco.exec
228K	jacoco.exec
```

**Pay attention on usage of latest released version of `quasar-core` which is `0.7.9` as of today.** With version `0.7.6` there is indeed exceptions.

JaCoCo agent is specified as a first one in order to follow our general recommendation about avoiding conflicts with other agents, even if in this case seems that order doesn't matter.

Warnings appearing in console seem to be the same with and without JaCoCo.
